api/controllers/Irc_EntryController.js

```diff
@@ -64,31 +67,36 @@ module.exports = {
 
   process_event: function (request_body) {
     if (request_body.operation === 'check in') {
-      return this.saveDetainee(request_body)
-        .then(this.saveEvent.bind(this, request_body));
+      return this.saveEvent(request_body);
     }
     throw new ValidationError('Unknown');
   },
 
-  getPID: (entity) => `${entity.centre}_${entity.person_id}`,
-
-  saveEvent: (request_body, detainee) =>
-    Events.create({
+  saveEvent: request_body => {
+    var event = {
+      person_id: request_body.person_id,
+      cid_id: request_body.cid_id,
+      gender: request_body.gender === 'm' ? 'male' : 'female',
+      nationality: request_body.nationality,
+      centre: request_body.centre,
       operation: request_body.operation,
-      timestamp: request_body.timestamp,
-      detainee: detainee
-    }),
+      timestamp: request_body.timestamp
+    };
+    var id = ModelHelpers.getPid.call(event);
 
-  saveDetainee: function (request_body) {
-    var person_id = this.getPID(request_body);
-    return Detainees.findOrCreate({
-      id: person_id
-    }, {
-      id: person_id,
-      cid_id: request_body.cid_id,
-      gender: request_body.gender,
-      centre: request_body.centre
-    });
+    return DetaineeEvent.create(event).then(() =>
+      DetaineeEvent.find({id: id}).then((detainees) =>
```

Comment:
I don't think you'll ever find an id here, since it's different to the one you just made in the event?